### ISSUE TITLE
After loading a file I get blank (all black) images instead in the viewer.

### ISSUE THREAD

**Issue by vincenzoml** [CBICA/CaPTk#654]
I have downloaded, installed and started CaPTk on Ubuntu 19.04 on a intel core i7 7700 machine with no GPU (except the integrated one). It "just doesn't work" out of the box. The program starts correctly, I load some images from the BRATS dataset, they are recognised correctly (also the modality, e.g. FLAIR), but the viewers stay black even if I drag my mouse around over them. The "Value" field changes when I drag the mouse, as if the image was there. Any suggestions?

**CaPTk Version**
1.7.2 

**Desktop (please complete the following information):**
 - OS: ubuntu
 - Version: 19.04

** CaPTk terminal window log **

Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Checking for compatible OpenGL - this will happen only once.
WARNING: Trying to run CaPTk GUI using software rendering - this might not work on all systems and in those cases, only the CLI will be available.
js: WARNING: cdn.mathjax.org has been retired. Check https://www.mathjax.org/cdn-shutting-down/ for migration tips.

**Comment by sarthakpati** [comment]
Thank you for your interest in CaPTk, @vincenzoml!

There are 2 issues here:

1. From the following message, it appears you are using Wayland: `Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway`. Is that correct? If yes, we have not built or tested CaPTk with Wayland support yet, though Qt has added it [[ref](https://wiki.qt.io/QtWayland)]. It will be available in a future release, though. In the meantime, is it possible for you to use Gnome desktop, instead?

2. This message: `WARNING: Trying to run CaPTk GUI using software rendering` basically says that a [compatible OpenGL version](https://cbica.github.io/CaPTk/FAQ.html#gs_FAQ_3) was not found. Please see the referenced link on ways you can work around it.

I will tag our GUI developer, @ashishsingh18 for the first issue.

**Comment by vincenzoml** [comment]
Hi, I tried xorg instead of wayland, but that does not change the issue (so the software renderer has a problem here). I will try to update my opengl drivers, although maybe this will be done by the next ubuntu release?

**Comment by sarthakpati** [comment]
I am not sure if and how Ubuntu packages their OpenGL drivers. You might need to do it by yourself.

**Comment by vincenzoml** [comment]
Hi, I first upgraded the opengl drivers via the PPA suggested at the link you provided. That bumped the opengl version slightly, but captk presented the same bug. Then I removed the ppa (using ppa-purge) and just upgraded to ubuntu 19.10. That did not solve the problem. Here is the "OpenGL" part of the output of "glxinfo".

OpenGL vendor string: Intel Open Source Technology Center
OpenGL renderer string: Mesa DRI Intel(R) HD Graphics 630 (Kaby Lake GT2) 
OpenGL core profile version string: 4.5 (Core Profile) Mesa 19.2.1
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 3.0 Mesa 19.2.1
OpenGL shading language version string: 1.30
OpenGL context flags: (none)
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.2 Mesa 19.2.1
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
OpenGL ES profile extensions:

**Comment by sarthakpati** [comment]
> OpenGL version string: 3.0 Mesa 19.2.1

I believe this issue (and a potential solution) is best illustrated in [this link](https://askubuntu.com/questions/888402/help-how-do-i-set-mesa-3-3-as-my-opengl-version-string):

```
MESA_GL_VERSION_OVERRIDE=4.5 #put the opengl version you want to use here
```

An addendum that provides more information is [here](https://askubuntu.com/questions/886548/making-sense-of-glxinfo-opengl-versions).

I will add both these to the FAQs.

**Comment by vincenzoml** [comment]
Setting MESA_GL_VERSION_OVERRIRE to 3.3 works fine. I would suggest you to do this automatically in a startup script and issue a warning, when MESA is detected; software renderer is very common in laptops, and not very many users will understand that this is the problem.

**Comment by sarthakpati** [comment]
> Setting MESA_GL_VERSION_OVERRIRE to 3.3 works fine

Excellent! 

We will see how best to work around this. Thank you for your feedback.
